### ISSUE TITLE
Support for lazy sessions

### ISSUE THREAD

**Issue by zkulbeda** [IlyaSemenov/grammy-scenes#7]
Bot loses the session with an entered scene when using lazy sessions. Official [documentation](https://grammy.dev/plugins/session.html#lazy-sessions) says, that developers should offer two ways of using session storage.
Example:
```typescript
import { Bot, Context, lazySession, LazySessionFlavor } from "grammy"
import { SceneSessionFlavor, ScenesFlavor } from "grammy-scenes"
import { scenes } from "./scenes"

type SessionData = SceneSessionFlavor & {}
export type BotContext = Context & LazySessionFlavor<SessionData> & ScenesFlavor

const bot = new Bot<BotContext>(process.env.BOT_TOKEN)
bot.use(
  lazySession({
    initial: () => ({}),
  })
)
bot.use(scenes.manager())
bot.command("start", async (ctx) => {
  await ctx.reply(`Welcome here.`)
  await ctx.scenes.enter("main")
})
bot.use(scenes)
bot.on("message", (ctx)=>ctx.reply("outside the scene"))
bot.start()
```
After sending /start and then sending some message, bot replies "outside the scene" message.

**Comment by IlyaSemenov** [comment]
There's not much win in using lazy session in scenes-centric bot because it **must** read the session on every request anyway. (As it needs to figure out the current scene/step and call the respective middleware).

But I agree that this as a missing feature/reasonable improve.

**Comment by IlyaSemenov** [comment]
Published in v10.1.0.
